### PR TITLE
using .innerHTML makes JSON reappear (closes #478)

### LINKED ISSUE #478: Adding a <script> element with the config information to the generated output
If a post-processor wants to do something with the generated HTML file, it would become very useful if the original config would be available in the file in a machine readable way. After all, many information (document status, short name, list of editors, etc) are available; currently the only way to get to those information is to extract them essentially scraping the content. An easy way would be to generate a `<script>` element to the file containing a JSON serialization of the config data. 

(My use case is a Python script I am finalizing for the generation of an EPUB3 version of a TR document.)

### PR DESCRIPTION
**Pull request by marcoscaceres**
For whatever reason, .innerText means that when .toString() is called within the output function, the inner text doesn't show up. However, using .innerHTML means that it does show up.

### CODE REVIEW

**Comment by marcoscaceres** [comment]
Tested in Safari, Chrome, and FF on Mac.

**Comment by marcoscaceres** [comment]
@iherman, I'm still not comfortable with this. Maybe it's better if you tell us exactly which fields you actually need for what you are doing?

**Comment by iherman** [comment]
@marcoscaceres I am not sure what you ask. Are you asking what I would use the respec config from my code if it was there?

If so: *at this moment* it would give me an easy way of finding, e.g., the publication date, the short name, the list of editors and authors, the document status and probably other things. All these information are in the initial respec setup but are thrown away; I must scrape the HTML output to find them. (Eg, find the latest URL relying on a particular class name, extract from the URL format whether it is a REC, or WD, or whatever, and extract the publication date.) I am sure other tools that may want to postprocess the HTML file may find other usage. 

(If it was *only* for my tool then I wouldn't have raised this feature request. After all, I have already gone through the trouble of doing it, although it is not future proof as it relies on a particular, and undocumented, structure of the final HTML.)

**Comment by marcoscaceres** [comment]
> On Sep 9, 2015, at 1:11 AM, Ivan Herman <notifications@github.com> wrote:
> 
> @marcoscaceres I am not sure what you ask. Are you asking what I would use the respec config from my code if it was there?
> 
> If so: at this moment it would give me an easy way of finding, e.g., the publication date, the short name, the list of editors and authors, the document status and probably other things.
> 

Ok, in addition to the above, I want to know the list that makes up "probably other things" (I can make it "all use-setable options, as per our documentation). We are vomiting a lot of ReSpec's internals into the JSON, and I would  prefer to only spit out the properties you + others need.

**Comment by iherman** [comment]
@marcoscaceres I would say that anything that the user can set should be visible to the output. We should not come up with a list of "visible" properties imho.

**Comment by marcoscaceres** [comment]
Sent from my iPhone

> On Sep 9, 2015, at 1:26 AM, Ivan Herman <notifications@github.com> wrote:
> 
> @marcoscaceres I would say that anything that the user can set should be visible to the output. We should not come up with a list of "visible" properties imho.
> 

The problem is that ReSpec abuses the user's configuration to also configure its own things. What I will do is snapshot the initial configuration (before ReSpec starts changing it) and return that. That should give you what you need without exposing ReSpec's internal config. Sound good?

**Comment by iherman** [comment]
> On 09 Sep 2015, at 07:34 , Marcos Caceres <notifications@github.com> wrote:
> 
> 
> 
> Sent from my iPhone
> 
> > On Sep 9, 2015, at 1:26 AM, Ivan Herman <notifications@github.com> wrote:
> >
> > @marcoscaceres I would say that anything that the user can set should be visible to the output. We should not come up with a list of "visible" properties imho.
> >
> 
> The problem is that ReSpec abuses the user's configuration to also configure its own things. What I will do is snapshot the initial configuration (before ReSpec starts changing it) and return that. That should give you what you need without exposing ReSpec's internal config. Sound good?

That sounds perfect. Thanks!

File: js/core/include-config.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ define(
           }
           return val;
         }
-        script.innerText = JSON.stringify(conf, confFilter, 2);
+        script.innerHTML = JSON.stringify(conf.initialUserConfig, confFilter, 2);
```
Comment (marcoscaceres) [inline js/core/include-config.js]: Note, innerText doesn't work... it doesn't output when respec calls .innerHTML on the root node.

**Comment by marcoscaceres** [comment]
@tobie r?

File: js/core/base-runner.js [hunk 2]
```diff
@@ -23,6 +23,16 @@
     ,   embedded = (top !== self)
     ;
     if (!("respecConfig" in window)) window.respecConfig = {};
+    // clone the initial user configuration.
+    try{
+        if(Object.assign){
```
Comment (tobie) [inline js/core/base-runner.js]: Style.

Comment (tobie) [inline js/core/base-runner.js]: Not sure what the convention us but iirc ReSpec has ws all over the place.

File: js/core/include-config.js (hunk 1 above)
Comment (tobie) [inline js/core/include-config.js]: I don't understand why this wouldn't work, but frontend dev is full of mysteries.

**Comment by tobie** [comment]
LGTM

File: js/core/base-runner.js (hunk 2 above)
Comment (marcoscaceres) [inline js/core/base-runner.js]: going to fix all that in #493.

**Comment by marcoscaceres** [comment]
@tobie thanks for the review + time!

**Pull request merged**